packages/next/client/route-announcer.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from 'react'
+import { useRouter } from './router'
+
+export function RouteAnnouncer() {
+  const router = useRouter()
+  const [routeAnnouncement, setRouteAnnouncement] = useState('')
+
+  const { asPath } = router
+
+  // Every time the path changes, announce the route change. The announcement will be prioritized by h1, then title
+  // (from metadata), and finally if those don't exist, then the pathName that is in the URL. This methodology is
+  // inspired by Marcy Sutton's accessible client routing user testing. More information can be found here:
+  // https://www.gatsbyjs.com/blog/2019-07-11-user-testing-accessible-client-routing/
+  useEffect(
+    () => {
+      let newRouteAnnouncement
+      const pageHeader = document.querySelector('h1')
+
+      if (pageHeader) {
+        newRouteAnnouncement = pageHeader.innerText
```

Comment:
Use `textContent` as the fallback to support Firefox <= 44.

```suggestion
        newRouteAnnouncement = pageHeader.innerText || pageHeader.textContent
```

Reply:
Thank you @shuding!! 🎉